packages/keystone/redis.js

```diff
@@ -1,12 +1,51 @@
 const IORedis = require('ioredis')
+const { get } = require('lodash')
 
 const conf = require('@open-condo/config')
 
 const { getLogger } = require('./logging')
 
+const READONLY_COMMANDS = ['bitcount', 'bitfield_ro', 'bitpos', 'dbsize', 'dump', 'eval_ro', 'evalsha_ro', 'exists', 'expiretime', 'fcall_ro', 'geodist', 'geohash', 'geopos', 'georadius_ro', 'georadiusbymember_ro', 'geosearch', 'get', 'getbit', 'getrange', 'hexists', 'hget', 'hgetall', 'hkeys', 'hlen', 'hmget', 'hrandfield', 'hscan', 'hstrlen', 'hvals', 'keys', 'lcs', 'lindex', 'llen', 'lolwut', 'lpos', 'lrange', 'mget', 'pexpiretime', 'pfcount', 'pttl', 'randomkey', 'scan', 'scard', 'sdiff', 'sinter', 'sintercard', 'sismember', 'smembers', 'smismember', 'sort_ro', 'srandmember', 'sscan', 'strlen', 'substr', 'sunion', 'touch', 'ttl', 'type', 'xlen', 'xpending', 'xrange', 'xread', 'xrevrange', 'zcard', 'zcount', 'zdiff', 'zinter', 'zintercard', 'zlexcount', 'zmscore', 'zrandmember', 'zrange', 'zrangebylex', 'zrangebyscore', 'zrank', 'zrevrange', 'zrevrangebylex', 'zrevrangebyscore', 'zrevrank', 'zscan', 'zscore', 'zunion']
+
+
+const PREFIX = process.cwd().split('/').pop() + ':'
 const REDIS_CLIENTS = {}
+
 const logger = getLogger('redis')
 
+if (get(conf, 'REDIS_FALLBACK_ENABLED', 'false') === 'true') {
+    const originalSendCommand = IORedis.prototype.sendCommand
+    IORedis.prototype.sendCommand = function (...args) {
+        const [command] = args
+
+        if (!READONLY_COMMANDS.includes(command.name)) {
+            return originalSendCommand.apply(this, args)
+        }
+
+        const prefixedKey = command.args[0]
+
+        return new Promise((resolve, reject) => {
+            originalSendCommand.apply(this, args).then((value) => {
+                if (value !== null || !prefixedKey.startsWith(PREFIX)) {
+                    resolve(value)
+                } else {
+                    const nonPrefixedKey = prefixedKey.substring(PREFIX.length)
+
+                    const otherCommand = new IORedis.Command(command.name, [nonPrefixedKey], 'utf-8')
```

Comment:
Are all commands have only 1 argument?

Reply:
If I understand correctly, the problem is fixed